### PR TITLE
WIP: Wire up webserver, container, build scripts, and move dirs

### PR DESCRIPTION
**Pull request by VnUgE**
This is an untested PR just getting started for initial review, I haven't had the time to test it yet.

So I'm not expecting you to merge, this is just a concept, if y'all want to keep the website code and server/container packages as part of our CI infra we can do that with a submodule and leave this repo alone. I think either would be good idea.

### Change summary
 - Moved the front-end code to a subdirectory called `frontend`
 - added [vnlib.webserver](https://www.vaughnnugent.com/Resources/Software/Modules/VNLib.Core?p=vnlib.webserver) specific configuration
- created a CI directory containing Taskfile.dev scripts that will
   - download vnlib.webserver, plugins and required native libraries
   - build the frontend package and copy to dir structure
   - create a container based deployment for Linux servers
- hard-coded server configuration 
- container (docker/podman) Dockerfile and docker-compose file to run it once deployed 
  - no volumes exposed
  - no environment variables needed
  - chiseled base without a shell or other security concerns 

### Todo
 - Actually finish the deployment as an artifact
 - Help get it wired up to Jenkins and whatever we want
 - Test it...
 -  Wire up to our infra

### CODE REVIEW

File: ci/docker-compose.yaml
Comment (Silberengel) [inline ci/docker-compose.yaml:1]: So it auto-restarts on a crash, or what does `restart: unless-stopped` mean?

**Review by Silberengel — APPROVED** [review]

File: ci/docker-compose.yaml
Comment (VnUgE) [reply ci/docker-compose.yaml:1]: For any reason a container may be stopped, unless it was stopped by the user that stated it, it should restart. System shutdown for example.

**Comment by VnUgE** [comment]
Also build scripts will be 100% cross platform so whatever remote agent running to build the container should be fine if you choose to build using my Task scripts. The build runs on the native shell (not in container)

**Comment by VnUgE** [comment]
If I'm not mistaken you'll probably want to run the build in a Docker container? If so maybe I should rename the CI directory to something else so we can run the scripts inside a container.

**Comment by VnUgE** [comment]
Scrapping this design, and moving server wrapping into our infra configuration.

**Pull request closed without merging**